src/org/omegat/core/machinetranslators/YandexCloudTranslate.java

```diff
@@ -0,0 +1,245 @@
+
+        String AIMToken = getAIMToken(oAuthToken);
+        if (AIMToken == null) {
+            return AIMErrorMessage;
+        }
+
+        StringBuilder requestBuilder = new StringBuilder();
+        requestBuilder.append("{");
+        requestBuilder.append("\"sourceLanguageCode\":\"");
+        requestBuilder.append(sLang.getLanguageCode().toLowerCase());
+        requestBuilder.append("\",");
+        requestBuilder.append("\"targetLanguageCode\":\"");
+        requestBuilder.append(tLang.getLanguageCode().toLowerCase());
+        requestBuilder.append("\",");
+
+        // HTML format keeps OmegaT tags intact
+        requestBuilder.append("\"format\": \"HTML\",");
+
+        requestBuilder.append("\"folderId\": \"");
+        requestBuilder.append(folderId);
+        requestBuilder.append("\",");
+
+        requestBuilder.append("\"texts\": [");
+        requestBuilder.append(JsonParser.quote(trText));
+        requestBuilder.append("]}");
+
+        Map<String, String> headers = new TreeMap<>();
+        headers.put("Authorization", "Bearer " + AIMToken);
+
+        String response;
+        try {
+            response = WikiGet.postJSON(TRANSLATE_URL, requestBuilder.toString(), headers);
+        } catch (WikiGet.ResponseError e) {
+            String errorMessage = extractErrorMessage(e.body);
+            if (errorMessage == null) {
+                errorMessage = OStrings.getString("MT_ENGINE_YANDEX_CLOUD_BAD_TRANSLATE_RESPONSE");
+            }
+            return errorMessage;
+        } catch (IOException e) {
+            return e.getLocalizedMessage();
+        }
+
+        String tr = extractTranslation(response);
+
+        if (tr == null) {
+            return "";
+        }
+
+        tr = cleanSpacesAroundTags(tr, trText);
+        putToCache(sLang, tLang, trText, tr);
+        return tr;
+    }
+
+    @SuppressWarnings("unchecked")
+    private String extractErrorMessage(final String json) {
+        Map<String, Object> rootNode;
+        try {
+            rootNode = (Map<String, Object>) JsonParser.parse(json);
+            return (String) rootNode.get("message");
+        } catch (Exception e) {
+            Log.logErrorRB(e, "MT_ENGINE_YANDEX_CLOUD_BAD_ERROR_REPORT");
+            return null;
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private String extractTranslation(final String json) {
+        try {
+            Map<String, Object> rootNode;
+            rootNode = (Map<String, Object>) JsonParser.parse(json);
+            List<Object> translationsList = (List<Object>) rootNode.get("translations");
+            Map<String, String> translationNode = (Map<String, String>) translationsList.get(0);
+            return translationNode.get("text");
+        } catch (Exception e) {
+            Log.logErrorRB(e, "MT_JSON_ERROR");
+            return OStrings.getString("MT_ENGINE_YANDEX_CLOUD_BAD_TRANSLATE_RESPONSE");
+        }
+    }
+
+    @Override
+    public boolean isConfigurable() {
```

Comment:
nitpick:

Overridden public methods should appear in the class body before private methods. 

Basically something like this:

```
public class MyClass extends SomeBaseClass {

   public static String PUBLIC_CONSTANT = "this should be first";
   private static String PRIVATE_CONSTANT = "then private constants";
   private String someInstanceVariable = "then non static intance variables";
   
   // then constructors
   public MyClass() {
        super();
   }
   
   @Override
   public void overriddenMethod() {
       // then public overridden methods
   } 

   private String helperMethod() {
       // then private methods
   }
}
```

### Source:

- the Java Language specification: https://docs.oracle.com/javase/specs/jls/se8/html/jls-8.html
- some checkstyle plugins: https://checkstyle.sourceforge.io/apidocs/com/puppycrawl/tools/checkstyle/checks/coding/DeclarationOrderCheck.html, https://checkstyle.org/styleguides/sun-code-conventions-19990420/CodeConventions.doc2.html#a1852
- SO: https://stackoverflow.com/questions/4668218/are-there-any-java-method-ordering-conventions